metaflow/plugins/kfp/kfp.py

```diff
@@ -979,6 +983,43 @@ def build_kfp_dag(
                     "get_workflow_uid"
                 )
 
+            s3_sensor_op = None
+            s3_sensor_deco = self.flow._flow_decorators.get("s3_sensor")
+            if s3_sensor_deco:
+                path = s3_sensor_deco.path
+                timeout_seconds = s3_sensor_deco.timeout_seconds
+                polling_interval_seconds = s3_sensor_deco.polling_interval_seconds
+                func = s3_sensor_deco.func
+
+                # see https://github.com/kubeflow/pipelines/pull/1946/files
+                # KFP does not support the serialization of Python functions directly. The KFP team took
+                # the approach of using base64 encoding + pickle. Pickle didn't quite work out
+                # in this case because pickling a function directly stores references to the function's path,
+                # which couldn't be resolved when the formatter function was unpickled within the running
+                # container. Instead, we took the approach of marshalling just the code of the formatter
+                # function, and reconstructing the function within the kf_s3_sensor.py code.
+                func_code_encoded = base64.b64encode(
+                    marshal.dumps(func.__code__)
+                ).decode("ascii")
+
+                s3_sensor_op = func_to_container_op(
+                    wait_for_s3_path,
+                    base_image="hsezhiyan/s3_sensor:1.0",
```

Comment:
why are you using your private image? Is there no public image you can use?

Reply:
This is a public image that anyone can pull.

Reply:
Is there anything special about this image? Does it contain any package / image layer that is necessary for s3_sensor? Please share if so. Otherwise maybe the image should be configurable instead of being hard coded, so that we can use some image from `aip-py` internally.

Reply:
@cloudw nothing special except for boto3 and botocore in these images. I created my own image because I couldn't find any official images online with just these packages (which would be good since they're lightweight). I found [some](https://hub.docker.com/r/jpbarto/boto3) created by other developers but I thought we can use one I created for more control (in case that image for example gets taken down).